cache/disk/disk_test.go

```diff
@@ -185,43 +185,88 @@ func TestCachePutWrongSize(t *testing.T) {
 }
 
 func TestCacheGetContainsWrongSize(t *testing.T) {
+
 	cacheDir := tempDir(t)
 	defer os.RemoveAll(cacheDir)
 	testCache := New(cacheDir, 100, nil)
 
-	content := "hello"
-	hash := hashStr(content)
-	correctSize := int64(len(content))
-
-	var err error
 	var found bool
+	var rdr io.ReadCloser
 
-	err = testCache.Put(cache.CAS, hash, correctSize, strings.NewReader(content))
+	err := testCache.Put(cache.CAS, contentsHash, contentsLength, strings.NewReader(contents))
 	if err != nil {
 		t.Fatal("Expected success", err)
 	}
 
-	found, _ = testCache.Contains(cache.CAS, contentsHash, correctSize+1)
+	found, _ = testCache.Contains(cache.CAS, contentsHash, contentsLength+1)
 	if found {
 		t.Error("Expected not found, due to size being different")
 	}
 
-	_, _, err = testCache.Get(cache.CAS, contentsHash, correctSize+1)
-	if err == nil {
-		t.Error("Expected error, due to size being different")
+	rdr, _, _ = testCache.Get(cache.CAS, contentsHash, contentsLength+1)
+	if rdr != nil {
+		t.Error("Expected not found, due to size being different")
 	}
 
 	found, _ = testCache.Contains(cache.CAS, contentsHash, -1)
 	if !found {
 		t.Error("Expected found, when unknown size")
 	}
 
-	_, _, err = testCache.Get(cache.CAS, contentsHash, -1)
-	if err != nil {
-		t.Error("Expected success, when unknown size")
+	rdr, _, _ = testCache.Get(cache.CAS, contentsHash, -1)
+	if rdr == nil {
+		t.Error("Expected found, when unknown size")
 	}
 }
 
+func TestCacheGetContainsWrongSizeWithProxy(t *testing.T) {
+
+	cacheDir := tempDir(t)
+	defer os.RemoveAll(cacheDir)
+	testCache := New(cacheDir, 100, new(proxyStub))
+
+	var found bool
+	var rdr io.ReadCloser
+
```

Comment:
Please add a comment here to point out that proxyStub contains the the digest {contentsHash, contentsLength}.

Reply:
Done